gpii/node_modules/matchMakerFramework/src/MatchMakerUtilities.js

```diff
@@ -377,6 +377,17 @@ var fluid = fluid || require("infusion"),
     };
 
     /**
+     * Filters preferences to contain only common terms and application specific settings for the solutionId
+     */
+    gpii.matchMakerFramework.utils.filterPreferencesForSolution = function (solutionId, allPreferences) {
+        var applicationString = "http://registry.gpii.net/applications/" + solutionId;
+
+        return fluid.remove_if(fluid.copy(allPreferences), function (value, preference) {
+            return (preference.indexOf("http://registry.gpii.net/common/") !== 0 && preference.indexOf(applicationString) !== 0);
```

Comment:
Unbelievably cryptic! Especially mysteriously conflating the case that the string is not found at all (index -1) and that it is found later in the string (index > 0).
Given we're in node or electron (a recent discussion confirmed that user-facing GPII web code is safe to exclude IE11) https://issues.gpii.net/browse/GPII-2471 so you can replace these with a preference.startsWith test, assuming that was what you meant : P